sql/src/planner.rs

```diff
@@ -397,46 +417,39 @@ impl<'a, P: MetaProvider> PlannerDelegate<'a, P> {
             )
         }
 
-        // Build primary key, the builder will check timestamp column is in primary key.
-        if let Some(idx) = primary_key_constraint_idx {
-            // If primary key is already provided, use that primary key.
-            if let TableConstraint::Unique { columns, .. } = &stmt.constraints[idx] {
-                for col in columns {
-                    let key_column = if TSID_COLUMN == col.value {
+        let timestamp_col_idx = timestamp_column_idx.context(RequireTimestamp)?;
+        // The key columns have been consumed.
+        for (idx, col) in stmt
+            .columns
+            .iter()
+            .enumerate()
+            .map(|(idx, column)| Ok((idx, parse_column(column)?)))
+            .collect::<Result<Vec<_>>>()?
+        {
+            if !primary_key_column_idxs.is_empty() {
```

Comment:
After insert_new_key_column & insert_new_normal_column is merged into one method, I think this place could be simplified further.